### DISCUSSION TITLE
central theme development with grapes js for various website builders

### DISCUSSION THREAD

**Discussion by linuxchico** [GrapesJS/grapesjs#4394] [Q&A]
Hi

I'm looking for a tool that I can use to develop a theme and then adapt it to different website builders. I need a graphical tool because I don't have much code knowledge. A little, but not much. The demo of grapes js looks good to me.

Now the central questions:

Can I use grapesjs to create a main theme that works for some of the important website builders? I'm thinking of wp, drupal, joomla, grav, hugo, nikola, pelican and maybe dokuwiki.

I have seen that each of these tools uses its own method to create themes, or offers theme builder plugins. But I only want to create my theme once, centrally and then adapt it to the tools.

Is that possible? If yes, how should I proceed? For which of the above tools grapesjs already has a solution (plugin, export function, adaptation function, etc.).

thanks for some tips!

**Comment by linuxchico** [comment]
Ok, no answers yet. Is it a bad sign?

On the website of grapes js I read:

"GrapesJS is an open-source, multi-purpose, Web Builder Framework which combines different tools and features with the goal to help you (or users of your application) to build HTML templates without any knowledge of coding."

But if I read the documentation, then this is full of codes!

Do I have too high expectations? I want a universal theme builder that works locally, as an application. I create my new theme, or edit an imported one. Then I export the theme with the help of an export function, which automatically adapts the theme to a selected website builder, i.e. structure and additional files, for example yaml. This way I can manage my theme 100% with grapes js and use it for different website builders, for example WP, Grav, Drupal, etc. 

Let me know, on grapes js can do something like that. Should it be possible to build something like this? I think it would be the first :-)

**Reply by artf** [reply]
GrapesJS is a framework to build builders, so what you're mentioning is definitely possible but not built-in. 
The documentation is not a user manual on how to use the demo but how to create your own builder, so a good knowledge of JS is definitely required.

**Reply by linuxchico** [reply]
@artf 

Hi Artur

Oh, I see. So there is no stand-alone application of grapesjs. That's a pity. But there are plugins for some CMS, for example, Drupal. There is also one for Grav, but it is already 4 years old. I will test it.

[https://github.com/zorca/grav-plugin-grapes-js-builder](url)

Grav is a flat file CMS and therefore very easy to "install". It also works locally if you have a current php version installed with a built-in server. If the plugin still works, I could easily create a theme locally with grapesjs, or edit an existing one. In addition, you can also generate static websites with Grav, with the plugin static generator.

This theme then works for Grav. If you would make some **export funcions** in grapesjs to adapt the theme to other CMS and static website builders, then you would have a kind of stand-alone version of grapesjs. It requires a zip with Grav core + Grav admin, then the grapesjs plugin, a very simple theme with only a few sections and a some sample content.

Anyone could download the zip, for example from github and unpack it locally, then start it. To log in you have to create an account. If you install it on a hoster, other people could also use it, create an account and develop a theme. With the **export functions**, you could automatically adapt the theme for the desired CMS, i.e. the folder structure and the additional files that the CMS requires.

In this way, there would be a standard theme builder with grapesjs, which can create a theme for the most important CSMs. As soon as this zip exists, other people could also install it on a hoster and there would be many websites where you can use grapesjs.

Ok, it's just an idea and I don't have the skills to realize it. But maybe others? Or you can turn grapesjs into a stand-alone application that you can install locally, including the **export functions**.

**Reply by artf** [reply]
Yeah unfortunately there is no real stand-alone application of grapesjs if not a few demos available from the homepage, but it's definitely something I'd like to add. At the moment I'm mainly focused on making our internal modules more customizable, the next step would be to leverage those improvements in a real stand-alone application.

**Comment by linuxchico** [comment]
Unfortunately, the plugin for Grav is pretty empty. It has only one function, it posted a sentence on the page.

**Comment by linuxchico** [comment]
Hi

I followed your instructions:

[https://grapesjs.com/docs/#quick-start](url)

Unfortunately, there are some problems here:

[https://github.com/artf/grapesjs-preset-webpage](url)

Downloading the zip works.

Usage does not contain any information about what to do with the code.

The code is faulty, for example 'gjs-preset-webpage'

Development
Clone the repository does not work, it says something about fingerprints

npm start does not seem to be a valid command

Issues are waiting for treatment.

Wiki is not available.

I think that the project could be very helpful for beginners if it were brought up to date. That's why it's so prominently mentioned in your documentary.

With the help of documentation of other projects, a user can find the way, but it would probably be better if he did not have to resort to other documentation.

[https://getuikit.com/docs/introduction#html-markup](url)

Please let someone from the team, who is competent, solve these problems soon. There are already enough projects on github that are poorly or not managed at all. Github seems to be a real project graveyard!

**Reply by ronaldohoch** [reply]
Hello!
As mentioned before, Grapesjs is a greate tool to build other tools, you will need to know a little about programming to build your own tool, as i and some other does.

Arthur don't work just for this project, but, you can fund an issue: https://issuehunt.io/r/artf/grapesjs

**Reply by linuxchico** [reply]
Thank you! I hope that the problems will be resolved soon. Otherwise, the link in the documentation should be deleted. Project management is imminent, or things quickly get out of hand.

**Who does what, by when, for how much?**

This question must be able to be answered in every project at any time, or the project is out of control. And if one or more of these 4 parts have a problem, then you have to correct with the others. Sometimes it is better to let part of the project die and use the forces to keep the rest alive. Simplicity is the top priority for user-friendliness. All-in-one packages, easy installation, basic theme for first attempts, modular expandability. And of course a documentation that corresponds to the current state of the product at all times. You do NOT publish a new version if the documentation is not ready.

No, I don't want to be instructive here, but when you start a project, you have to learn something about project management first, or you will sooner or later get shipwrecked.

all the best with grapesjs!
